src/code_16c8e0.c

```diff
@@ -1,4 +1,137 @@
 #include "common.h"
+#include "battle/battle.h"
+#include "script_api/battle.h"
+
+f32 D_802809F0 = 0.0f;
+s8 D_802809F4 = 0;
+s8 D_802809F5 = 0;
+s16 D_802809F6 = -1;
+s16 D_802809F8 = 0;
+u16 D_802809FA = 0;
+
+s32 D_802809FC[] = {
+    0x80291FA8, 0x80291FD0, 0x80291FF8, 0x80292020, 0x80292048, 0x80292070, 0x80292098, 0x802920C0, 0x802920E8, 0x80292110, 0x00000000, 0x00000000, 0x00000000,
+};
+
+s32 D_80280A30 = 0xFF;
+
+Script BtlPutPartnerAway = SCRIPT({
+    DispatchEvent(256, 62);
+    parallel {
+        SI_VAR(0) = 1.0;
+        loop 10 {
+            SetActorScale(256, SI_VAR(0), SI_VAR(0), 1.0);
+            SI_VAR(0) -= 0.1005859375;
+            sleep 1;
+        }
+    }
+    EnablePartnerBlur();
+    PlaySoundAtActor(0, 14);
+    GetActorPos(0, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    SI_VAR(1) += 25;
+    SetActorJumpGravity(256, 1.0);
+    SetGoalPos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    JumpToGoal(256, 10, 0, 0, 1);
+    DisablePartnerBlur();
+});
+
+Script BtlBringPartnerOut = SCRIPT({
+    parallel {
+        SI_VAR(0) = 0.1005859375;
+        loop 20 {
+            SetActorScale(256, SI_VAR(0), SI_VAR(0), 1.0);
+            SI_VAR(0) += 0.05078125;
+            sleep 1;
+        }
+        SetActorScale(256, 1.0, 1.0, 1.0);
+    }
+    PlaySoundAtActor(0, 13);
+    GetGoalPos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    SetActorJumpGravity(256, 1.0);
+    if (SI_VAR(1) == 0) {
+        JumpToGoal(256, 20, 0, 0, 1);
+    } else {
+        JumpToGoal(256, 20, 0, 0, 1);
+    }
+    GetActorPos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    ForceHomePos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+});
+
+u8 D_80280CE0[] = { 0, 0, 0, 0 };
```

Comment:
file padding? Also, why u8?

Reply:
No,
```
glabel add_cam_zoom
/* 17CE94 8024E5B4 3C028028 */  lui       $v0, %hi(D_80280CE0)
/* 17CE98 8024E5B8 80420CE0 */  lb        $v0, %lo(D_80280CE0)($v0)
```

Reply:
The latter 3 bytes are possibly file padding, but I'd rather wait for us to be certain

Reply:
agr33d